lib/platform/linux.c

```diff
@@ -64,7 +64,7 @@ struct switchtec_linux {
 
 const char *platform_strerror(void)
 {
-	return "Success";
+	return strerror(errno);
```

Comment:
This isn't quite valid. strerror() also applies on windows, it just doesn't cover all the cases and we need to fall back on the windows platform errors.

Reply:
Sorry, I didn't get your point. There's a platform_strerror() implementation on the windows platform, but it's using Windows system APIs, not strerror().

Reply:
You still need to call strerror() on windows and all other platforms. strerror() is standard to C not to linux. Malloc calls, for example, always set errno. Windows also needs to call it's platform specific error handler because all the syscalls the interface with the operating system have a different error facility.

So moving the strerror into the linux specific platform is wrong. It needs to remain in the general case.

Reply:
Sorry, seems strerror() (return string describing error number) standard to Linux, its counterpart on Windows here is "FormatMessage function", (Formats a message string, while FORMAT_MESSAGE_FROM_SYSTEM is specified, an an application can pass the result of the GetLastError function to retrieve the message text for a system-defined error)

Reply:
No. strerror() is *not* standard to Linux. It's standard to C and applies to all standard C functions. Linux also uses it heavily for system calls. Windows does not use it for system calls and instead uses FormatMessage but windows still uses it for all standard C functions (like malloc). 

So, once again, moving the strerror into the linux specific platform is wrong. Don't do it.